source/river/src/files.rs

```diff
@@ -0,0 +1,96 @@
+use std::ops::{Deref, DerefMut};
+
+use pandora_module_utils::{
+    pingora::SessionWrapper, OneOrMany, RequestFilter, RequestFilterResult,
+};
+use pingora::{server::Server, upstreams::peer::HttpPeer};
+use pingora_core::Result;
+use pingora_proxy::{ProxyHttp, Session};
+use static_files_module::{StaticFilesConf, StaticFilesHandler};
+
+use crate::{config::internal::FileServerConfig, populate_listners};
+
+pub fn river_file_server(
+    conf: FileServerConfig,
+    server: &Server,
+) -> Box<dyn pingora::services::Service> {
+    let fsconf = StaticFilesConf {
+        root: conf.base_path,
+        canonicalize_uri: true,
+        index_file: OneOrMany::from(Vec::new()),
```

Comment:
`OneOrMany` type is an implementation detail that you probably don’t want to have in your code. `Vec::new().into()` or `Default::default()` will do as well.

In fact, it looks like you really want `StaticFilesConf {root: conf.base_path, ..Default::default()}` here.

Reply:
Thanks! I overlooked this choice, and will switch to your proposed initialization.